signinwithapplebutton/src/main/java/com/willowtreeapps/signinwithapplebutton/SignInWithAppleService.kt

```diff
@@ -49,8 +49,7 @@ class SignInWithAppleService(
     See the Sign In With Apple Javascript SDK for comparison:
     https://developer.apple.com/documentation/signinwithapplejs/configuring_your_webpage_for_sign_in_with_apple
     */
-    internal fun buildAuthenticationAttempt(): AuthenticationAttempt {
-        val state = UUID.randomUUID().toString()
+    internal fun buildAuthenticationAttempt(state: String = UUID.randomUUID().toString()): AuthenticationAttempt {
         val authenticationUri = Uri
```

Comment:
I'm not sure I agree with making the state an argument. We need it to be unique per attempt. What's your goal?

Reply:
Goal was just testability yeah, since it's an internal function wasn't too worried about the client setting it wrong.

Reply:
That's true… and from a testing perspective, the single equality comparison of the entire object is less brittle in the case that we change the contents of the object. OK, I'm down.